src/app/components/weapon-list/weapon-list.component.html

```diff
@@ -1,75 +1,24 @@
-      <div id="weapon-list-container" class="flex-column" *ngIf="listItems">
-        <div
-          class="weapon-group-list-item flex-column background-gray-2"
-          *ngFor="let listItem of listItems | keyvalue"
-        >
-          <div
-            class="weapon-group-list-item-image-container flex-row flex-center flex-center-items background-gray-3"
-          >
-            <img
-              *ngIf="listItem.value.mainImage"
-              src="{{ listItem.value.mainImage }}"
-            />
-            <h3 *ngIf="listItem.value.mainLabel">
-              {{ listItem.value.mainLabel }}
-            </h3>
-          </div>
-          <div class="weapon-items-container flex-row flex-center">
-            <a
-              class="weapon-item-container flex-row background-gray-1"
-              *ngFor="let weaponItem of listItem.value.weapons"
-              routerLink="{{ '/weapons/' + weaponItem.Id }}"
-            >
-              <div
-                class="weapon-item-title-container flex-column flex-center flex-center-items"
-              >
-                <h4>{{ weaponItem?.Name }}</h4>
-                <moe-rarity [value]="weaponItem.Rarity"></moe-rarity>
-              </div>
+<div class="list-page-header">
+  <h1>Weapons</h1>
+</div>
+
+<div class="list-page-content">
+  <moe-loading-image *ngIf="!listItems; else elseBlock"></moe-loading-image>
 
-              <div class="weapon-item-image-container-1">
-                <div class="weapon-item-image-container-2">
-                  <div
-                    class="weapon-item-image-container-3 flex-row flex-center flex-center-items"
-                    [ngStyle]="{
-                      'background-image': 'url(' + weaponItem.Image + ')'
-                    }"
-                  >
-                    <div class="weapon-item-overlay-background-container">
-                      <div
-                        class="weapon-item-overlay"
-                        [ngStyle]="{
-                          'background-image': 'url(' + weaponItem?.Image + ')'
-                        }"
-                      ></div>
-                    </div>
-                    <div
-                      class="weapon-item-overlay-content flex-column flex-center flex-center-items position-absolute"
-                    ></div>
-                  </div>
-                </div>
-              </div>
-            </a>
-          </div>
-        </div>
+  <ng-template #elseBlock>
+    <div class="weapon-filter-sort-stub"></div>
+    <div class="weapon-group-section" *ngFor="let listItem of listItems | keyvalue">
+      <div class="weapon-group-section-header">
+        <img *ngIf="listItem.value.mainImage" src="{{ listItem.value.mainImage }}" />
+        <h3 *ngIf="listItem.value.mainLabel">
+          {{ listItem.value.mainLabel }}
+        </h3>
+      </div>
+      <div class="weapon-items-list">
+        <moe-weapon-card *ngFor="let weapon of listItem.value.weapons"
```

Comment:
https://github.com/angular/angular/issues/9209 😪

Reply:
this is the nerdiest thing I've ever read

Reply:
🥓🤳